core/iwasm/fast-jit/fe/jit_emit_memory.c

```diff
@@ -526,27 +526,179 @@ jit_compile_op_memory_grow(JitCompContext *cc, uint32 mem_idx)
 }
 
 #if WASM_ENABLE_BULK_MEMORY != 0
+static bool
+wasm_init_memory(WASMModuleInstance *inst, uint32 seg_idx, uint32 len,
+                 uint32 mem_offset, uint32 data_offset)
+{
+    WASMMemoryInstance *mem_inst;
+    WASMDataSeg *data_segment;
+    uint32 mem_size;
+    uint8 *mem_addr, *data_addr;
+
+    /* if d + n > the length of mem.data */
+    mem_inst = inst->memories[0];
+    mem_size = mem_inst->cur_page_count * mem_inst->num_bytes_per_page;
+    if (mem_size < mem_offset || mem_size - mem_offset < len)
+        goto out_of_bounds;
+
+    /* if s + n > the length of data.data */
+    bh_assert(seg_idx < inst->module->data_seg_count);
+    data_segment = inst->module->data_segments[seg_idx];
+    if (data_segment->data_length < data_offset
+        || data_segment->data_length - data_offset < len)
+        goto out_of_bounds;
+
+    mem_addr = mem_inst->memory_data + mem_offset;
+    data_addr = data_segment->data + data_offset;
+    bh_memcpy_s(mem_addr, mem_size - mem_offset, data_addr, len);
+
+    return true;
+out_of_bounds:
+    wasm_set_exception(inst, "out of bounds memory access");
+    return false;
+}
+
 bool
-jit_compile_op_memory_init(JitCompContext *cc, uint32 seg_index)
+jit_compile_op_memory_init(JitCompContext *cc, uint32 seg_idx)
 {
+    JitReg len, mem_offset, data_offset, res;
+    JitReg args[5] = { 0 };
+
+    POP_I32(len);
+    POP_I32(data_offset);
+    POP_I32(mem_offset);
+
+    res = jit_cc_new_reg_I32(cc);
+    args[0] = get_module_inst_reg(cc->jit_frame);
+    args[1] = NEW_CONST(I32, seg_idx);
+    args[2] = len;
+    args[3] = mem_offset;
+    args[4] = data_offset;
+
+    if (!jit_emit_callnative(cc, wasm_init_memory, res, args,
+                             sizeof(args) / sizeof(args[0])))
+        goto fail;
```

Comment:
Should check return value `res` to return as soon as possible when exception is thrown, ref to:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/dev/fast_jit/core/iwasm/fast-jit/fe/jit_emit_function.c#L170-L177